src/panoptes/pocs/camera/camera.py

```diff
@@ -418,6 +444,21 @@ def take_exposure(self,
 
         assert filename is not None, self.logger.error("Must pass filename for take_exposure")
 
+        if not self.can_take_internal_darks:
+            if dark and self.filterwheel and self.filterwheel.dark_position:
+                # Can't take internal darks, but do have an opaque filter for that.
+                msg = "Taking dark exposure using filter '" + \
+                    f"{self.filterwheel.filter_name(self.filterwheel.dark_position)}'."
+                self.logger.debug(msg)
+                self.filterwheel.move_to_dark_position(blocking=True)
```

Comment:
Changed all of this to be based entirely on `try .. except` instead of testing truthiness of lots of attributes. Needed to modify both `move_to_dark_position` and `move_to_light_position` to raise exceptions (`error.NotFound`) instead of merely logging a warning about the absence of the set dark position or saved last light position.